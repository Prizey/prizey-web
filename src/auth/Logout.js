import React from 'react'
import { SignOut } from 'croods-auth'
import { Redirect } from '@reach/router'

export default ({ children, className, ...props }) => (
  <SignOut
    {...props}
    render={({ destroy }) => (
      <span
        className={className}
        style={{ color: 'white', cursor: 'pointer' }}
        onClick={destroy}
      >
        {children}
      </span>
    )}
    renderDestroyed={() => <Redirect to="/sign-in" noThrow />}
  />
)
