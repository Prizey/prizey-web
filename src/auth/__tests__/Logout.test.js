import React from 'react'
import renderer from 'react-test-renderer'

import Logout from '../Logout'

jest.mock('croods-auth', () => ({
  SignOut: props => (
    <div {...props}>
      SignOut - {props.render({ destroy: jest.fn() })}-{props.renderDestroyed()}
    </div>
  ),
}))

jest.mock('@reach/router', () => ({
  Redirect: props => <div {...props}>Redirect</div>,
}))

it('renders correctly', () => {
  const tree = renderer.create(<Logout>Logout</Logout>).toJSON()
  expect(tree).toMatchSnapshot()
})
