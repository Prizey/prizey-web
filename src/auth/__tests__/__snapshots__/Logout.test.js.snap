// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  renderDestroyed={[Function]}
>
  SignOut - 
  <span
    onClick={[MockFunction]}
    style={
      Object {
        "color": "white",
        "cursor": "pointer",
      }
    }
  >
    Logout
  </span>
  -
  <div
    noThrow={true}
    to="/sign-in"
  >
    Redirect
  </div>
</div>
`;
